src/Raven.Server/Documents/Replication/AbstractReplicationLoader.cs

```diff
@@ -0,0 +1,541 @@
+                case InternalReplication _:
+                case ExternalReplication _:
+                    authorizationInfo = null;
+                    return _server.Server.Certificate.Certificate;
+
+                case PullReplicationAsSink sink:
+                    authorizationInfo = new TcpConnectionHeaderMessage.AuthorizationInfo
+                    {
+                        AuthorizeAs = sink.Mode switch
+                        {
+                            PullReplicationMode.HubToSink => TcpConnectionHeaderMessage.AuthorizationInfo.AuthorizeMethod.PullReplication,
+                            PullReplicationMode.SinkToHub => TcpConnectionHeaderMessage.AuthorizationInfo.AuthorizeMethod.PushReplication,
+                            PullReplicationMode.None => throw new ArgumentOutOfRangeException(nameof(node), "Replication mode should be set to pull or push"),
+                            _ => throw new ArgumentOutOfRangeException("Unexpected replication mode: " + sink.Mode)
+                        },
+                        AuthorizationFor = sink.HubName
+                    };
+
+                    if (sink.CertificateWithPrivateKey == null)
+                        return _server.Server.Certificate.Certificate;
+
+                    var certBytes = Convert.FromBase64String(sink.CertificateWithPrivateKey);
+                    return new X509Certificate2(certBytes, sink.CertificatePassword, X509KeyStorageFlags.Exportable | X509KeyStorageFlags.MachineKeySet);
+
+                default:
+                    throw new ArgumentException($"Unknown node type {node.GetType().FullName}");
+            }
+        }
+
+        public void EnsureNotDeleted(string node)
+        {
+            using (_server.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+            using (ctx.OpenReadTransaction())
+            {
+                using (var rawRecord = _server.Cluster.ReadRawDatabaseRecord(ctx, _databaseName))
+                {
+                    if (rawRecord != null && rawRecord.DeletionInProgress.ContainsKey(node))
+                    {
+                        throw new OperationCanceledException($"The database '{_databaseName}' on node '{node}' is being deleted, so it will not handle replications.");
+                    }
+                }
+            }
+        }
+
+        protected virtual void AssertValidConnection(IncomingConnectionInfo connectionInfo)
+        {
+            if (_server.IsPassive())
+            {
+                throw new InvalidOperationException(
+                    $"Cannot accept the incoming replication connection from {connectionInfo.SourceUrl}, because this node is in passive state.");
+            }
+        }
+
+        protected virtual DynamicJsonValue GetInitialRequestMessage(ReplicationLatestEtagRequest replicationLatestEtagRequest,
+            ReplicationLoader.PullReplicationParams replParams = null)
```

Comment:
can we avoid the `ReplicationLoader.PullReplicationParams replParams = null` here ?